fluentd/templates/statefulset.yaml

```diff
@@ -0,0 +1,171 @@
+          {{- end }}
+          name: {{ .Chart.Name }}
+          ports:
+            {{- if .Values.metrics.enabled }}
+            - name: prometheus
+              containerPort: {{ .Values.metrics.port }}
+              hostPort: {{ .Values.metrics.port }}
+              protocol: TCP
+            {{- end }}
+            {{- if .Values.fluentd.http.enabled }}
+            - name: http
+              containerPort: {{ .Values.fluentd.http.port }}
+              hostPort: {{ .Values.fluentd.http.port }}
+              protocol: TCP
+            {{- end }}
+            {{- if .Values.fluentd.forward.enabled }}
+            - name: forward
+              containerPort: {{ .Values.fluentd.forward.port }}
+              hostPort: {{ .Values.fluentd.forward.port }}
+              protocol: TCP
+            {{- end }}
+            {{- with .Values.extraPort }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          {{- if and .Values.readinessProbe.enabled .Values.fluentd.http.enabled }}
+          readinessProbe:
+            httpGet:
+              path: / tfluentd.healthcheck?json=%7B%22ping%22%3A+%22pong%22%7D
+              port: http
+            initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.readinessProbe.successThreshold }}
+            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          volumeMounts:
+            {{- if .Values.persistence.enabled }}
+            - name: {{ template "fluentd.fullname" . }}
+              mountPath: /var/log
+            {{- end }}
+            {{- range $name, $tmpl := .Values.configmaps }}
+            {{- if eq $name "fluent.conf" }}
+            - name: {{ $name | replace "." "-" | replace "_" "-" | lower }}
+              mountPath: /fluentd/etc/{{ $name }}
+              subPath: {{ $name }}
+            {{- else }}
+            - name: {{ $name | replace "." "-" | replace "_" "-" | lower }}
+              mountPath: /fluentd/etc/config.d/{{ $name }}
+              subPath: {{ $name }}
+            {{- end }}
+            {{- end }}
+            {{- range $name, $secret := .Values.secrets }}
+            - name: {{ $name | replace "." "-" | replace "_" "-" | lower }}
+              mountPath: /secure/{{ $name }}
+              subPath: {{ $name }}
+            {{- end }}
+            {{- with .Values.extraVolumeMounts }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+      imagePullSecrets:
+        {{- toYaml .Values.imagePullSecrets | nindent 8 }}
+      nodeSelector:
+        {{- toYaml .Values.nodeSelector | nindent 8 }}
```

Comment:
The `nodeSelector` will be deprecated at somewhere, so I don't think it's necessary.
https://github.com/kubernetes/kubernetes/issues/82331

Reply:
@cw-sakamoto 
In `helm create`, it is added by default to the chart you create.
I also consider it unnecessary, but since it's supported by the latest HELM, shouldn't it be left?

Reply:
Of course, there is no problem with it as it is.